src/messages.ts

```diff
@@ -38,6 +38,77 @@ class Key {
  */
 export type LoaderFunction = (locale: string) => Messages;
 
+/**
+ * Different message file content parsers. This allows us to have js, json, and md. Maybe we will eventually support yaml, etc.
+ */
+type FileParser = (filePath: string, fileContents: string) => Map<string, AnyJson>;
+
+const markdownLoader: FileParser = (filePath: string, fileContents: string): Map<string, AnyJson> => {
+  const map = new Map<string, AnyJson>();
```

Comment:
`Map<string, string>`?

Reply:
Well, you can have nested JsonMaps and JsonArrays. For example, 

```
{
  "myKey": {
    "mySubKey: ["item1", "item2"]
  }
}
```

I could do something like the following but it didn't really seem worth it at the time.

```
type MessageContainer = Map<string, string | string[] | MessageContainer>
```

I can make the change if you like that better.

Reply:
Nah it's fine as is